testing/typescript/data/stores.ts

```diff
@@ -20,6 +27,24 @@ export async function infersTItemFromComplexLoadResult() {
   const value: { a: number } = result;
 }
 
+export async function loadReturnRightTypeIfLoadDefinedInOptions() {
+  const dataItem = { id: 1, value: true };
+  const store = new CustomStore({
+    // eslint-disable-next-line @typescript-eslint/require-await
+    load: async () => ({
+      data: [dataItem],
+      totalCount: 1,
+    }),
+  });
+
+  const loadResult = await store.load();
+  const actualItem: (typeof loadResult) extends ResolvedData<infer Item> ? Item : never = ANY;
+
+  notAny(actualItem);
+  notNever(actualItem);
+  assertType<typeof dataItem>(toAssertion(actualItem));
```

Comment:
Let's add one more case:

```suggestion
  notAny(loadItem);
  notNever(loadItem);
  assertType<typeof dataItem>(toAssertion(loadItem));

  const byKeyItem = await store.byKey(ANY);
  notAny(byKeyItem);
  notNever(byKeyItem);
  assertType<typeof dataItem>(toAssertion(byKeyItem));```